File: Admin.NET/Admin.NET/Admin.NET.Application/Entity/CmsPage.cs
using SqlSugar;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Admin.NET.Application.Entity;

/// <summary>
/// CMS Page（N Page + Tempalte page）
/// </summary>
[SugarTable("CmsPage")]
public class CmsPage: EntityBaseTenantDel
{

    /// <summary>
    /// Page type: "template" or "page"
    /// </summary>
    [SugarColumn(Length = 50)]
    public string Pagetype { get; set; }
    [SugarColumn]
    public long TemplateId { get; set; }

    [SugarColumn]
    public long? Pid { get; set; }
    /// <summary>
    /// SEO title
    /// </summary>
    [SugarColumn(Length = 255)]
    public string? Title { get; set; }

    /// <summary>
    /// SubeTitle
    /// </summary>
    [SugarColumn(Length = 255)]
    public string SubTitle { get; set; }

    /// <summary>
    /// Status
    /// </summary>
    [SugarColumn]
    public int Status { get; set; }

    /// <summary>
    /// Request path
    /// </summary>
    [SugarColumn(Length = 255)]
    public string? RequestPath { get; set; }

    /// <summary>
    /// Real path
    /// </summary>
    [SugarColumn(Length = 255)]
    public string? RealPath { get; set; }

    /// <summary>
    /// Whether the page is dynamic
    /// </summary>
    [SugarColumn]
    public bool IsDynamic { get; set; } = false;

 

    /// <summary>
    /// SEO description
    /// </summary>
    [SugarColumn]
    public string? Description { get; set; }

    /// <summary>
    /// Keywords
    /// </summary>
    [SugarColumn]
    public string? Keywords { get; set; }

    /// <summary>
    /// Canonical URL
    /// </summary>
    [SugarColumn(Length = 255)]
    public string? CanonicalUrl { get; set; }

    /// <summary>
    /// Robots setting
    /// </summary>
    [SugarColumn(Length = 50)]
    public string? Robots { get; set; }

    /// <summary>
    /// Open Graph title
    /// </summary>
    [SugarColumn(Length = 255)]
    public string? OgTitle { get; set; }

    /// <summary>
    /// Open Graph image URL
    /// </summary>
    [SugarColumn(Length = 255)]
    public string? OgImage { get; set; }

    /// <summary>
    /// Open Graph type
    /// </summary>
    [SugarColumn(Length = 50)]
    public string? OgType { get; set; }
}

File: Admin.NET/Web/src/api/content/cmsPage.ts
import {useBaseApi} from '/@/api/base';

// page接口服务
export const useCmsPageApi = () => {
	const baseApi = useBaseApi("cmsPage");
	return {
		// 分页查询page
		page: baseApi.page,
		// 查看page详细
		detail: baseApi.detail,
		// 新增page
		add: baseApi.add,
		// 更新page
		update: baseApi.update,
		// 删除page
		delete: baseApi.delete,
		// 批量删除page
		batchDelete: baseApi.batchDelete,
		// 导出page数据
		exportData: baseApi.exportData,
		// 导入page数据
		importData: baseApi.importData,
		// 下载page数据导入模板
		downloadTemplate: baseApi.downloadTemplate,
	}
}

// page实体
export interface CmsPage {
	// 主键Id
	id: number;
	// Pagetype
	pagetype?: string;
	// TemplateId
	templateId?: number;
	// Title
	title: string;
	// SubeTitle
	subeTitle?: string;
	// Status
	status?: number;
	// RequestPath
	requestPath: string;
	// RealPath
	realPath: string;
	// IsDynamic
	isDynamic?: boolean;
	// Description
	description: string;
	// Keywords
	keywords: string;
	// CanonicalUrl
	canonicalUrl: string;
	// Robots
	robots: string;
	// OgTitle
	ogTitle: string;
	// OgImage
	ogImage: string;
	// OgType
	ogType: string;
	// TenantId
	tenantId: number;
	// IsDelete
	isDelete?: boolean;
	// DeleteTime
	deleteTime: string;
	// CreateTime
	createTime: string;
	// UpdateTime
	updateTime: string;
	// CreateUserId
	createUserId: number;
	// CreateUserName
	createUserName: string;
	// UpdateUserId
	updateUserId: number;
	// UpdateUserName
	updateUserName: string;
}